src/app/data_harvester.rs

```diff
@@ -325,10 +329,12 @@ impl DataCollector {
                 #[cfg(target_os = "linux")]
                 {
                     processes::get_process_data(
+                        &self.sys,
                         &mut self.prev_idle,
                         &mut self.prev_non_idle,
                         &mut self.pid_mapping,
                         self.use_current_cpu_total,
+                        self.per_core_percentage,
```

Comment:
`get_process_data` fires after we get CPU data from heim (for now). We could just leverage that information and pass in the number of cores directly as opposed to having to pull in `System` and doing a refresh above.

Reply:
What if `use_cpu` is false?

Reply:
And if you were talking about using the length of `cpu_data`, the new code would also need to look at `show_average_cpu`, because when `show_average_cpu` is true, `get_cpu_data_list` would prepend the average usage to the vec.

Reply:
> What if `use_cpu` is false?

Hmm, fair enough. Could probably just use [heim](https://docs.rs/heim/0.1.0-rc.1/heim/cpu/fn.logical_count.html) beforehand then.

Reply:
What about non-linux paths where `sys` is already passed around? Should I change these as well for consistency or keep the use of `sys.cpus().len()` in these paths?

Reply:
Done.